.pre-commit-config.yaml

```diff
@@ -9,4 +9,21 @@ repos:
         # types_or: [ python, pyi, jupyter ] # lets you include notebooks in linting
       # Run the formatter.
       - id: ruff-format
-        # types_or: [ python, pyi, jupyter ] # lets you include notebooks in formatting
\ No newline at end of file
+        # types_or: [ python, pyi, jupyter ] # lets you include notebooks in formatting
+  - repo: https://github.com/PyCQA/bandit
+    rev: 1.8.6
+    hooks:
+      - id: bandit
+        args: [--config=bandit.yml, -b, bandit_baseline.json]  # will run bandit on the entire project
+  - repo: https://github.com/gitleaks/gitleaks
+    rev: v8.27.2
+    hooks:
+      - id: gitleaks
+  - repo: https://github.com/fabasoad/pre-commit-snyk
+    rev: v1.0.2
+    hooks:
+      - id: snyk-test
+        args:
+          - --snyk-args=--file=poetry.lock
+          - --snyk-args=--package-manager=pip
+          - --snyk-args=--fail-on=upgradable
```

Comment:
This is a policy decision for the project. This change fixes 2 upgradable dependency issues found in the initial scan.

I view this as a sane starter policy, but welcome maintainer/reviewer input here.